demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -1191,6 +1191,8 @@ static MQTTStatus_t prvWaitForPacket( MQTTContext_t * pxMQTTContext,
         /* Event callback will set #usPacketTypeReceived when receiving appropriate packet. This
          * will wait for at most mqttexamplePROCESS_LOOP_TIMEOUT_MS. */
         xMQTTStatus = MQTT_ProcessLoop( pxMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
```

Comment:
These demos are often used as a starting point when developing a new application.  Why does a `vTaskDelay()` need to be here?  Would all applications using this library follow this pattern, or are we adding this due to a special case in one particular application?